Source/WebCore/rendering/RenderElement.cpp

```diff
@@ -2123,7 +2123,12 @@ bool RenderElement::createsNewFormattingContext() const
 
 bool RenderElement::establishesIndependentFormattingContext() const
 {
-    return isFloatingOrOutOfFlowPositioned() || hasPotentiallyScrollableOverflow() || style().containsLayout() || paintContainmentApplies();
+    return isFloatingOrOutOfFlowPositioned() || hasPotentiallyScrollableOverflow() || style().containsLayout() || paintContainmentApplies() || isBlockLevelElementWithBlockStepSizing();
+}
+
+bool RenderElement::isBlockLevelElementWithBlockStepSizing() const
```

Comment:
Not sure we need `isBlockLevelElement` in the name, the spec makes it clear that block-step-size only applies to block-level containers.

In fact I wonder if this should be a method on `RenderStyle`, something like `RenderStyle::effectiveBlockStepSize()` that returns `none` when it's inline level. 

@alanbaradlay Any opinions here?

Reply:
please check "9.2.1 Block-level elements and block boxes" for proper naming
[visuren](https://www.w3.org/TR/CSS22/visuren.html) or just simply look at LFC (all these names have been properly resolved in LFC long time ago)

Reply:
@anttijk may have some opinions on the RenderStyle part.

Reply:
the problem with effective* pattern in general is that it's highly error prone as people could easily overlook it and go with the not-effective (what does that even mean) call (e.g. blockStepSize() in this case)...also you need to know what it actually means to decided whether you need to call this or the other one. Not a big fan.

Reply:
I just ended up removing the function and extracted out the logic into the method I was using it at. If I really end up needing this in multiple locations we can consider adding it back with perhaps a different name or approach